File: clients/python/src/hiero_proxy/resources/files.py
"""Files resource."""

from __future__ import annotations

from typing import TYPE_CHECKING, Any

from hiero_proxy.resources.validation import require_id

if TYPE_CHECKING:
    from hiero_proxy.client import AsyncHieroProxyClient, HieroProxyClient


class FilesResource:
    """File operations — create, read, update, delete."""

    def __init__(self, client: HieroProxyClient | AsyncHieroProxyClient) -> None:
        self._client = client

    def create(self, contents: str, expiration_time_epoch_second: int | None = None) -> Any:
        body: dict[str, Any] = {"contents": contents}
        if expiration_time_epoch_second is not None:
            body["expirationTimeEpochSecond"] = expiration_time_epoch_second
        return self._client.post("/api/v1/files", json=body)

    def get_contents(self, file_id: str) -> Any:
        fid = require_id(file_id, "file_id")
        return self._client.get(f"/api/v1/files/{fid}/contents")

    def get_info(self, file_id: str) -> Any:
        fid = require_id(file_id, "file_id")
        return self._client.get(f"/api/v1/files/{fid}")

    def update_contents(self, file_id: str, contents: str) -> Any:
        fid = require_id(file_id, "file_id")
        return self._client.put(f"/api/v1/files/{fid}/contents", json={"contents": contents})

    def update_expiration(self, file_id: str, expiration_time_epoch_second: int) -> Any:
        fid = require_id(file_id, "file_id")
        return self._client.put(f"/api/v1/files/{fid}/expiration", json={"expirationTimeEpochSecond": expiration_time_epoch_second})

    def delete(self, file_id: str) -> Any:
        fid = require_id(file_id, "file_id")
        return self._client.delete(f"/api/v1/files/{fid}")

File: clients/javascript/src/resources/files.ts
import type { HieroProxyClient } from "../client";
import type {
  FileCreateRequest,
  FileUpdateContentsRequest,
  FileUpdateExpirationRequest,
  FileCreatedResponse,
  FileContentsResponse,
  FileInfoResponse,
} from "./files.types";
import type { SuccessResponse } from "./types";
import { requireId } from "./validation";

/**
 * Files resource — create, read, update, and delete files on the Hiero File Service.
 */
export class FilesResource {
  constructor(private readonly client: HieroProxyClient) {}

  /**
   * Create a file on the ledger.
   * @param contents Base64-encoded file contents.
   * @param expirationTimeEpochSecond Optional expiration timestamp.
   */
  async create(contents: string, expirationTimeEpochSecond?: number): Promise<FileCreatedResponse> {
    const body: FileCreateRequest = { contents, expirationTimeEpochSecond };
    return this.client.post("/api/v1/files", body);
  }

  /**
   * Get file contents (Base64-encoded).
   */
  async getContents(fileId: string): Promise<FileContentsResponse> {
    const id = requireId(fileId, "fileId");
    return this.client.get(`/api/v1/files/${id}/contents`);
  }

  /**
   * Get file info (size, deleted status, expiration).
   */
  async getInfo(fileId: string): Promise<FileInfoResponse> {
    const id = requireId(fileId, "fileId");
    return this.client.get(`/api/v1/files/${id}`);
  }

  /**
   * Update file contents.
   * @param contents New Base64-encoded contents.
   */
  async updateContents(fileId: string, contents: string): Promise<SuccessResponse> {
    const id = requireId(fileId, "fileId");
    const body: FileUpdateContentsRequest = { contents };
    return this.client.put(`/api/v1/files/${id}/contents`, body);
  }

  /**
   * Update file expiration time.
   */
  async updateExpiration(fileId: string, expirationTimeEpochSecond: number): Promise<SuccessResponse> {
    const id = requireId(fileId, "fileId");
    const body: FileUpdateExpirationRequest = { expirationTimeEpochSecond };
    return this.client.put(`/api/v1/files/${id}/expiration`, body);
  }

  /**
   * Delete a file.
   */
  async delete(fileId: string): Promise<SuccessResponse> {
    const id = requireId(fileId, "fileId");
    return this.client.delete(`/api/v1/files/${id}`);
  }
}
